### PR TITLE
Fix issue "local variable 'detail' referenced before assignment"

### PR DESCRIPTION
**Pull request by fangge1212**
Signed-off-by: Fangge Jin <fjin@redhat.com>

### CODE REVIEW

**Comment by fangge1212** [comment]
@dzhengfy 
Please help me review this pr

**Comment by dzhengfy** [comment]
@fangge1212 , please describe how this problem happens firstly.
And could you help paste some test results as this change is not very few.

**Comment by fangge1212** [comment]
Updated commit message with "how the problem happens"

**Comment by fangge1212** [comment]
Without this pr, when there is exception during test, it reports :

\# avocado run --vt-type libvirt --vt-machine-type q35 virsh.migrate.there_dname.without_cpu_hotplug.compat_migration.non_compat_migration.without_postcopy --vt-connect-uri qemu:///system
JOB ID     : 92b18eef3a9fd5bba1b87fca7fd32682f3ce16cb
JOB LOG    : /root/avocado/job-results/job-2019-07-11T16.04-92b18ee/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.migrate.there_dname.without_cpu_hotplug.compat_migration.non_compat_migration.without_postcopy: **ERROR: local variable 'detail' referenced before assignment** (398.19 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 401.28 s

**Comment by fangge1212** [comment]
With this pr, when there is exception during test, it can report exact error:

\# avocado run --vt-type libvirt --vt-machine-type q35 virsh.migrate.there_dname.without_cpu_hotplug.compat_migration.non_compat_migration.without_postcopy --vt-connect-uri qemu:///system
JOB ID     : 26dbffda7fab273c517db314780d93fbb9d266bf
JOB LOG    : /root/avocado/job-results/job-2019-07-11T14.47-26dbffd/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.migrate.there_dname.without_cpu_hotplug.compat_migration.non_compat_migration.without_postcopy: **ERROR: Error occurred. \n<class 'virttest.remote.LoginAuthenticationError'>: Got password prompt twice**    (output: "\nPermission denied, please try again.\nroot@192.168.124.70's password: ") (394.99 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 398.53 s

File: libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py [hunk 1]
```diff
@@ -1304,75 +1304,68 @@ def check_vm_state_dest():
...
-    vm.destroy()
-    vm.undefine()
-    orig_config_xml.define()
-
-    # cleanup xml created during memory hotplug test
-    if mem_hotplug:
-        if os.path.isfile(mem_xml):
-            data_dir.clean_tmp_files()
-            logging.debug("Cleanup mem hotplug xml")
+        test.error("Error occurred. \n%s: %s" % (detail.__class__, detail))
```
Comment (kylazhang) [inline libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py]: No need to catch exceptions for above all 3 if no need to do other operations.

Comment (fangge1212) [reply libvirt/tests/src/virsh_cmd/domain/virsh_migrate.py]: Updated

**Comment by kylazhang** [comment]
approve when fix the check fail

**Comment by fangge1212** [comment]
Move the result check part before "finally", please review again.
@dzhengfy

**Pull request merged**